File: Practice/app/src/main/java/com/example/practice/MainActivity.java
package com.example.practice;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.activity.result.ActivityResultLauncher;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    ActivityResultLauncher<String[]> mPermissionResultLauncher;

    private boolean isReadPermissionGranted = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        Button createPost = findViewById(R.id.add_post);
        Button requestPermission = findViewById(R.id.request_permission);

        createPost.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, AddPostActivity.class));
            }
        });

    }
}

File: PracticeApp/app/src/main/java/com/example/practiceapp/MainActivity.kt
package com.example.practiceapp

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val addPost: Button = findViewById(R.id.add_post)
        addPost.setOnClickListener {
            val intent = Intent(this, AddPostActivity::class.java)
            // start your next activity
            startActivity(intent)
        }
    }
}